react/components/ProductAssemblyOptions/InputValue/OptionsInputValue.tsx

```diff
@@ -13,71 +23,90 @@ const BOX_OPTIONS_HANDLES = [
   'optionsInputValueOptionBoxContainer',
 ] as const
 
-const DropdownOptions: FC<Props> = ({ inputValueInfo }) => {
-  const [state, onChange] = useInputValue(inputValueInfo)
+export type OptionDisplay = 'select' | 'box' | 'smart'
+
+type Options = {
+  label: string
+  value: string
+}
+
+type InnerProps = {
+  label: string
+  options: Options[]
+  inputId: string
+}
+
+type Props = {
+  optionsDisplay?: OptionDisplay
+  inputValueInfo: OptionsInputValue
+}
+
+const DropdownOptions: FC<InnerProps> = ({ label, options, inputId }) => {
+  const [state, onChange] = useInputValue(inputId)
   const handles = useCssHandles(DROPDOWN_OPTIONS_HANDLES)
 
   const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     const { value } = e.target
+
     onChange({ value })
   }
 
-  const options = useMemo(() => {
-    return inputValueInfo.domain.map((option) => ({
-      value: option,
-      label: option,
-    }))
-  }, [inputValueInfo.domain])
-
   return (
     <div className={`${handles.optionsInputValueDropdown} mb4`}>
       <Dropdown
         value={state}
         onChange={handleChange}
-        label={inputValueInfo.label}
+        label={label}
         options={options}
       />
     </div>
   )
 }
 
-const BoxOptions: FC<Props> = ({ inputValueInfo }) => {
-  const [state, onChange] = useInputValue(inputValueInfo)
+const BoxOptions: FC<InnerProps> = ({ label, options, inputId }) => {
+  const [state, onChange] = useInputValue(inputId)
   const handles = useCssHandles(BOX_OPTIONS_HANDLES)
 
   const handleKeyDown = (event: React.KeyboardEvent) => {
     const selected = state as string
-    const options = inputValueInfo.domain
-    const selectedIndex: number = options.indexOf(selected)
+    const selectedIndex: number = options.findIndex((i) => i.value === selected)
 
     switch (event.key) {
       case 'ArrowRight': {
         const count = options.length
         const nextSelectedIndex = (selectedIndex + 1) % count
-        const newValue = options[nextSelectedIndex]
-        onChange({ value: newValue })
+        const newOption = options[nextSelectedIndex]
+
+        onChange({ value: newOption.value })
         break
       }
+
       case 'ArrowLeft': {
         const count = options.length
         // Linter is triggering a false positive here :/
         // eslint-disable-next-line @typescript-eslint/restrict-plus-operands
```

Comment:
Just a comment not related to this PR: I feel like this rule triggers a lot of false positives

Reply:
the comment is outdated. removed it.